WallsAndHoles/meshview.cpp

```diff
@@ -122,6 +118,18 @@ void MeshView::cleanUp()
     mContext = nullptr;
 }
 
+void MeshView::cameraActivated(AbstractTool *tool, QString)
+{
+    if (mCamera)
+        disconnect(mCamera);
+
+    mCamera = static_cast<AbstractMeshViewCamera *>(tool);
```

Comment:
Ouch! Downcasting is evil. Have you considered making a wrapper tool, e.g.
```cc
class ToolWithActivate : public AbstractTool {
public:
   template<typename Func>
   ToolWithActivate(AbstractTool *otherTool, Func activateFunc);
   // store activateFunc as an std::function<void()>
   
   void activate() override {
      mActivateFunc();
      mOtherTool->activate();
   }
}
```

And then giving MeshView an `addCamera()` method that does this:
```cc
void MeshView::addCamera(AbstractCamera *cam)
{
   mTools->registerTool(new ToolWithActivate(cam, [this, cam] () {
      if (mCamera)
         disconnect(mCamera);
      mCamera = cam;
      connect(mCamera, ...);
      update();
   }
}
```

Then if you make mTools private (and hidden from MeshViewContainer too), then you could:
1) Add non-camera tools to MeshView (since now the camera-connecting code is only run for tools added with addCamera()).
2) Prevent the future possibility of someone modifying MeshViewContainer and forgetting that `mMeshView->mTools->registerTool()` can only be called with camera parameters, causing a static_cast crash.